app/src/test/java/net/squanchy/search/view/ItemsAdapterTest.java

```diff
@@ -0,0 +1,497 @@
+package net.squanchy.search.view;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import net.squanchy.schedule.domain.view.Event;
+import net.squanchy.search.SearchResults;
+import net.squanchy.search.view.SpeakerAdapter.ViewTypeId;
+import net.squanchy.speaker.domain.view.Speaker;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import static net.squanchy.schedule.domain.view.EventFixtures.anEvent;
+import static net.squanchy.speaker.domain.view.SpeakerFixtures.aSpeaker;
+import static org.fest.assertions.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+
+@RunWith(Enclosed.class)
+public class ItemsAdapterTest {
+
+    public static class TotalItemsCount extends BaseTest {
+
+        @Test
+        public void givenEmptySearchResults_whenGettingTotalItemsCount_thenReturnsZero() {
+            givenEmptySearchResults();
+
+            int totalItemsCount = itemsAdapter.totalItemsCount();
+
+            assertThat(totalItemsCount).isZero();
+        }
+
+        @Test
+        public void givenSearchResultsWithOnlyEvents_whenGettingTotalItemsCount_thenReturnsOneMoreThanTheNumberOfSpeakers() {
+            givenSearchResultsWith(ANY_THREE_EVENTS, NO_SPEAKERS);
+
+            int totalItemsCount = itemsAdapter.totalItemsCount();
+
+            assertThat(totalItemsCount).isEqualTo(4);      // 4 = 1 header + 3 events
+        }
+
+        @Test
+        public void givenSearchResultsWithOnlySpeakers_whenGettingTotalItemsCount_thenReturnsOneMoreThanTheNumberOfSpeakers() {
+            givenSearchResultsWith(NO_EVENTS, ANY_TWO_SPEAKERS);
+
+            int totalItemsCount = itemsAdapter.totalItemsCount();
+
+            assertThat(totalItemsCount).isEqualTo(3);      // 3 = 1 header + 2 speakers
+        }
+
+        @Test
+        public void givenSearchResultsWithEventsAndSpeakers_whenGettingTotalItemsCount_thenReturnsTwoMoreThanTheNumberOfEventsAndSpeakers() {
+            givenSearchResultsWith(ANY_THREE_EVENTS, ANY_TWO_SPEAKERS);
+
+            int totalItemsCount = itemsAdapter.totalItemsCount();
+
+            assertThat(totalItemsCount).isEqualTo(7);      // 7 = 1 header + 3 events + 1 header + 2 speakers
+        }
+
+        @Test
+        public void givenSearchResultsWithEventsAndSpeakers_whenGettingViewTypeAt_thenReturns() {
```

Comment:
I think you forgot to name this test 😆 

Also, I think it should go in `ViewType`